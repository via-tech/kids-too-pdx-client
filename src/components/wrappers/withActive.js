import React from 'react';
import PropTypes from 'prop-types';

export const withActive = Component => {
  function WithActive(props) {
    const { role, history } = props;

    if(role === 'org' || role === 'admin') {
      return <Component {...props} />;
    } else {
      return history.push('/activate');
    }
  }

  WithActive.propTypes = {
    role: PropTypes.string,
    history: PropTypes.object
  };

  return WithActive;
};
