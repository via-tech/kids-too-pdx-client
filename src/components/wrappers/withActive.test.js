import React from 'react';
import renderer from 'react-test-renderer';
import { withActive } from './withActive';

describe('With active wrapper', () => {
  const Component = withActive(() =>
    <div>
      <h1>The Main Component</h1>
    </div>
  );
  
  it('snapshots withActive with inactive org', () => {
    const props = {
      role: 'inactive',
      history: {
        push(route) {
          return route;
        }
      }
    };

    const tree = renderer.create(
      <Component {...props} />
    );

    expect(tree).toMatchSnapshot();
  });

  it('snapshots withActive with activated org', () => {
    const props = {
      role: 'org',
      history: {
        push(route) {
          return route;
        }
      }
    };

    const tree = renderer.create(
      <Component {...props} />
    );

    expect(tree).toMatchSnapshot();
  });
});
