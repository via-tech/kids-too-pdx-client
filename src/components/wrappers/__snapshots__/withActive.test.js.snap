// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With active wrapper snapshots withActive with activated org 1`] = `
<div>
  <h1>
    The Main Component
  </h1>
</div>
`;

exports[`With active wrapper snapshots withActive with inactive org 1`] = `"/activate"`;
